packages/rest/src/actions/middleware-action.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright IBM Corp. 2018. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  config,
+  Constructor,
+  InvocationResult,
+  Next,
+  Provider,
+} from '@loopback/context';
+import {RequestHandler} from 'express';
+import {HttpContext} from '../types';
+import {BaseRestAction} from './base-action';
+
+/**
+ * Wrap an Express middleware handler as RestAction
+ */
+export class MiddlewareAction extends BaseRestAction {
+  constructor(private requestHandler: RequestHandler) {
+    super();
+  }
+
+  async intercept(ctx: HttpContext, next: Next) {
+    return new Promise<InvocationResult>((resolve, reject) => {
+      try {
+        if (this.requestHandler.length < 3) {
+          // The express middleware does not call `next`
+          this.requestHandler(ctx.request, ctx.response, () => {});
+          resolve(next());
+          return;
+        }
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        this.requestHandler(ctx.request, ctx.response, (err?: any) => {
+          if (err) reject(err);
+          else resolve(next());
```

Comment:
IIUC, when the middleware handler function accepts `next` callback but does not call it, then the interceptor does not call `next` either. I find that problematic, because it breaks the assumptions that interceptors can rely on `await next()` mechanism to run code before and after request processing.

Example middleware:

```ts
function static(req, res, next) {
  const asset = path.join(assetsRoot, req.path);
  if (fs.existsSync()) {
    res.sendFile(assert)
  } else {
    next();
  }
}
// or also
const static = new MiddlewareAction(require('serve-static'), assetsRoot);
```

Example interceptor (a sequence action):

```ts
async function logRequest(ctx, next) {
  const start = process.hrtime();
  await next();
  const delta = process.hrtime(start);
  console.log(
    '%s %s %s %s ms',
    ctx.req.method, 
    ctx.req.path, 
    ctx.res.statusCode, 
    // time in milliseconds
    Math.round(delta[0]*1e3 + delta[1]/1e6) 
  );
}
```

In this example, no log messages are printed for requests served as static assets.